experimental/SetPartitions/src/Util.jl

```diff
@@ -123,3 +123,46 @@ function _add_partition_top_bottom(vector::Vector{Dict{Int, Set{T}}}, p::T) wher
 
     return vector
 end
+
+"""
+    simplify_operation(partition_sum::Vector{Tuple{S, T}}) where { S <: AbstractPartition, T <: RingElement }
+
+Simplify the vector representation of a `LinearSetPartition` in terms of associativity.
+"""
+function simplify_operation(partition_sum::Vector{Tuple{S, T}}) where { S <: AbstractPartition, T <: RingElement }
+
+    partitions = Dict{S, T}()
+
+    for (i1, i2) in partition_sum
+        
+        if iszero(i2)
+            deleteat!(partition_sum, findall(x -> x == (i1, i2), partition_sum))
+            continue
+        end
+        
+        if !(i1 in keys(partitions))
+            partitions[i1] = i2
+        else
+            deleteat!(partition_sum, findall(x -> x == (i1, i2), partition_sum))
+            deleteat!(partition_sum, findall(x -> x == (i1, get(partitions, i1, -1)), partition_sum))
+            push!(partition_sum, (i1, get(partitions, i1, -1) + i2))
+            partitions[i1] = get(partitions, i1, -1) + i2
+        end
```

Comment:
This function seems overly complicated to me. An honestly, I don't understands what it wants to achieve, in particular the associavity in the docstring I cannot relate with what's written here. Could you maybe explain here what this is supposed to do, and then we can see how to improve this?

Reply:
Hi, first of all thanks for your feedback! :)
The reason why you are not understanding it could be because I confused associativity with distributivity and made the function way more complicated than it should be... so thanks for pointing it out. I improved the code and corrected the docstring. I hope that this can clear things up!